src/backend/x11/mod.rs

```diff
@@ -0,0 +1,563 @@
+                                        // to match libinput.
+                                        key: key_release.detail as u32 - 8,
+                                        count: key_counter_val,
+                                        state: KeyState::Released,
+                                    },
+                                }),
+                                &mut event_window,
+                            );
+                        }
+                    }
+
+                    x11::Event::MotionNotify(motion_notify) => {
+                        if motion_notify.event == window.inner {
+                            // Use event_x/y since those are relative the the window receiving events.
+                            let x = motion_notify.event_x as f64;
+                            let y = motion_notify.event_y as f64;
+
+                            callback(
+                                Input(InputEvent::PointerMotionAbsolute {
+                                    event: X11MouseMovedEvent {
+                                        time: motion_notify.time,
+                                        x,
+                                        y,
+                                        size: window.size(),
+                                    },
+                                }),
+                                &mut event_window,
+                            )
+                        }
+                    }
+
+                    x11::Event::ResizeRequest(resized) => {
+                        if resized.window == window.inner {
+                            let size: Size<u16, Logical> = (resized.width, resized.height).into();
+
+                            // Intentionally drop the lock on the size mutex incase a user
+                            // requests a resize or does something which causes a resize
+                            // inside the callback.
+                            {
+                                *window.size.lock().unwrap() = size;
+                            }
+
+                            (callback)(X11Event::Resized(size), &mut event_window);
+                        }
+                    }
+
+                    x11::Event::ClientMessage(client_message) => {
+                        if client_message.data.as_data32()[0] == window.atoms.WM_DELETE_WINDOW // Destroy the window?
+                            && client_message.window == window.inner // Same window
+                        {
+                            (callback)(X11Event::CloseRequested, &mut event_window);
+                        }
+                    }
+
+                    x11::Event::Expose(expose) => {
+                        // TODO: We would ideally use this to determine damage and render more efficiently that way.
+                        //
+                        // Although there is an Expose with damage event somewhere?
```

Comment:
I don't understand the question, but I'll try to answer it anyway:

A expose event contains a rectangle that was exposed (x, y, width, height). If a non-rectangular area is exposed (e.g. Two other windows partially cover your window and now your window is brought to front. Thus, these two rectangles are exposed), then two Expose events are sent. The first one contains `.count == 1` to indicate that another event will follow.

(And if there were four rectangles, then the first one would have `count = 3`, the next one `count = 2` etc...)